src/android/io/sentry/SentryCordova.java

```diff
@@ -1,47 +1,370 @@
+
+        break;
+      case "clearBreadcrumbs":
+        clearBreadcrumbs(callbackContext);
+        break;
+      case "getStringBytesLength":
+        String payload = args.getString(0);
+
+        int length = getStringBytesLength(payload);
+
+        callbackContext.sendPluginResult(new PluginResult(Status.OK, length));
+
+        break;
+      case "setTag":
+        String tagKey = args.getString(0);
+        String tag = args.getString(1);
+
+        setTag(tagKey, tag, callbackContext);
+
+        break;
+      case "setContext":
+        String contextKey = args.getString(0);
+        JSONObject contextObject = null;
+        if (!args.isNull(1)) {
+          contextObject = args.getJSONObject(1);
+        }
+
+        setContext(contextKey, contextObject, callbackContext);
+
+        break;
+      case "setExtra":
+        String extraKey = args.getString(0);
+        String extra = args.getString(1);
+
+        setExtra(extraKey, extra, callbackContext);
+
+        break;
+      default:
+        callbackContext.sendPluginResult(new PluginResult(Status.ERROR, "not implemented"));
+        break;
+      }
+
+    } catch (JSONException e) {
+      logger.log(Level.SEVERE, "Error parsing JSON from native bridge");
+
+      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
+
+      return false;
+    } catch (Exception e) {
+      logger.log(Level.SEVERE, "Error occurred on native bridge: ", e);
+
+      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
+
+      return false;
+    }
+
+    return true;
+  }
+
+  private void startWithOptions(final JSONObject jsonOptions, final CallbackContext callbackContext) {
+    SentryAndroid.init(this.cordova.getActivity().getApplicationContext(), options -> {
+      try {
+        if (jsonOptions.has("dsn") && jsonOptions.getString("dsn") != null) {
           String dsn = jsonOptions.getString("dsn");
+          logger.info(String.format("Starting with DSN: '%s'", dsn));
+          options.setDsn(dsn);
+        } else {
+          // SentryAndroid needs an empty string fallback for the dsn.
+          options.setDsn("");
+        }
+        if (jsonOptions.has("debug") && jsonOptions.getBoolean("debug")) {
+          options.setDebug(true);
+        }
+        if (jsonOptions.has("maxBreadcrumbs")) {
+          options.setMaxBreadcrumbs(jsonOptions.getInt("maxBreadcrumbs"));
+        }
+        if (jsonOptions.has("environment") && jsonOptions.getString("environment") != null) {
+          options.setEnvironment(jsonOptions.getString("environment"));
+        }
+        if (jsonOptions.has("release") && jsonOptions.getString("release") != null) {
+          options.setRelease(jsonOptions.getString("release"));
+        }
+        if (jsonOptions.has("dist") && jsonOptions.getString("dist") != null) {
+          options.setDist(jsonOptions.getString("dist"));
+        }
+        if (jsonOptions.has("enableAutoSessionTracking")) {
```

Comment:
on iOS for Flutter and RN we needed a fix as the Plugin init. after the App's lifecycle, what's about this one?

Reply:
Yes this will need it too, good call!

Reply:
@marandaneto Can I have a clarification of what you were referring to with this again, I misunderstood and mixed it up with what we had for sessions on iOS.

Reply:
its basically to try out the release health for cordova using the native SDKs, if it works ootb or not.
For Flutter, we needed to fix something for iOS, see https://github.com/getsentry/sentry-dart/pull/274/files

Reply:
Ah I was confused, through there was also something for Android. Will follow this up in a separate PR then @marandaneto thanks for reminding me.